internal/resource.go

```diff
@@ -83,6 +84,24 @@ func GetUrltoTempFile(u string, ctx context.Context) (string, error) {
 }
 
 func (l *Resource) Download(dir string, mode os.FileMode, ctx context.Context) error {
+	// Check if a cache URL exists to use Artifactory first
+	if l.CacheUri != "" {
+		localName := ""
+		if l.Filename != "" {
+			localName = l.Filename
+		} else {
+			localName = path.Base(l.Urls[0])
+		}
+		resPath := filepath.Join(dir, localName)
+
+		err := downloadFromArtifactory(ctx, l.CacheUri, l.Integrity, resPath, mode)
+		if err == nil {
+			return nil
+		}
+		// Return the error and let Lock show the warning
+		return err
```

Comment:
So... you're failing here if the download from the cache doesn't work. This is no what we said we should do; we should:
- try to download from the cache
- if it works, return success
- if it doesn't work, issue a warning and continue the execution, trying to download the resource from the other urls